src/Controller/MapController.php

```diff
@@ -17,27 +18,55 @@
  */
 class MapController extends AbstractController
 {
+
+    //Affiche les entité "finness" sur la carte en fonction du tri
+    //et toutes les entités si il n'y a pas de tri
+
     /**
      * @Route("/", name="finess_map", methods={"GET"})
      * @param FinessRepository $finessRepository
      * @return Response
      */
-    public function index(FinessRepository $finessRepository): Response
+    public function index(FinessRepository $finessRepository, Request $request, StrainRepository $strainRepo): Response
     {
         $form = $this->createForm(
             MapSelectionType::class,
             null,
-            [//'action' => $this->generateUrl('route'),
-            'method'=>Request::METHOD_GET]
+            ['method'=>Request::METHOD_GET]
         );
 
-        $fine = $finessRepository->findAll();
+        $form->handleRequest($request);
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+            // si le form est vide le controlleur renvoie tt les établissements
+            if ($data['microOrganisme']== 'Laisser vide') {
+                $fine = $finessRepository->findAll();
+            //sinon il affiche les établissement dans lesquel la souche selectionnée a été trouvée
+            } else {
+                //un établissement contient plusieurs souche, récupération des souches qui ont le nom envoyé par le form
+                $strains = $strainRepo->findBy(['microOrganisme' => $data['microOrganisme']]);
```

Comment:
You probably should have a more strict validation than just "Not 'Laisser vide'"